lib/LaTeXML/Package/TeX.pool.ltxml

```diff
@@ -5479,22 +5479,27 @@ DefMathI('\biguplus', undef, "\x{228E}",
   meaning   => 'symmetric-difference',
   mathstyle => \&doVariablesizeOp,
   font      => { size => 'Big' });
-DefConstructorI('\limits', undef, sub {
-    my $node = $_[0]->getElement;
-    $_[0]->setAttribute($_[0]->getLastChildElement($node) || $node, scriptpos => 'mid'); });
-DefConstructorI('\nolimits', undef, sub {
-    my $node = $_[0]->getElement;
-    $node = $_[0]->getLastChildElement($node) || $node;
-    $node->removeAttribute('scriptpos'); });    # default is 'post', so we can just remove the attrib.
-DefConstructorI('\displaylimits', undef, sub {
-    my ($document, %props) = @_;
-    my $node = $_[0]->getElement;
-    $node = $_[0]->getLastChildElement($node) || $node;
-    if (($props{mathstyle} || 'text') eq 'display') {
-      $document->setAttribute($node, scriptpos => 'mid'); }
-    else {
-      $node->removeAttribute('scriptpos'); } },
-  properties => sub { (mathstyle => LookupValue('font')->getMathstyle); });
+DefConstructorI('\limits', undef, '',
+  afterDigest => sub { mergeLimits('mid'); },
+  properties  => { isSpace => 1 });
+DefConstructorI('\nolimits', undef, '',
+  afterDigest => sub { mergeLimits('post'); },
+  properties  => { isSpace => 1 });
+DefConstructorI('\displaylimits', undef, '',
+  afterDigest => sub {
+    mergeLimits((($_[1]->getProperty('mathstyle') || '') eq 'display' ? 'mid' : 'post')); },
+  properties => { isSpace => 1 });
+
+sub mergeLimits {
+  my ($pos) = @_;
+  for (my $i = scalar(@LaTeXML::LIST) - 1 ; $i >= 0 ; $i--) {
```

Comment:
it's been a while since I saw an old-school for loop! Hope there are no off-by-one errors in there :> Looks like it could also work with:

```perl
for my $box(reverse(@LaTeXML::LIST)) {
  ... }
```

but your version is certainly the faster code.